### PR TITLE
Polls created from templates

### PR DESCRIPTION
**Pull request by tpkelly**

### CODE REVIEW

**Comment by tpkelly** [comment]
Poll options are only cloned on the front end, not the back end.

File: VotingApplication/VotingApplication.Web.Api/Scripts/Create.js [hunk 1]
```diff
@@ -112,30 +102,11 @@
...
-                ko.utils.registerEventHandler($hoverTarget, "mouseover", showElement);
-                ko.utils.registerEventHandler($hoverTarget, "mouseout", hideElement);
-                hideElement();
-            }
-        }
-
         self.initialise = function () {
-            self.populateTemplates();
-            setupTooltips();
+            var defaultExpiryDate = moment().add(30, 'minutes');
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web.Api/Scripts/Create.js]: Do we want this here? I feel this should be set when the user submits a poll rather than when they enter the page.

File: VotingApplication/VotingApplication.Web.Api/Scripts/Manage.js [hunk 2]
```diff
@@ -135,9 +195,16 @@
             });
         }
 
+        self.isSignedIn = function () {
+            return signedIn();
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web.Api/Scripts/Manage.js]: Is there no way to pass this into the constructor? Seems bad to have this floating as a global function.

File: VotingApplication/VotingApplication.Web.Api/Scripts/Signin.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        self.password = ko.observable("");
+
+        self.newUsername = ko.observable("");
+        self.newPassword = ko.observable("");
+        self.confirmPassword = ko.observable("");
+
+        var checkPasswordFormat = function (password) {
+            // Checks for password with uppercase, lowercase, digits and punctuation.
+            // AccountController fails the password if any of these are missing
+            var passwordRegex = /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[!@#$&*]).*$/;
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web.Api/Scripts/Signin.js]: Personally, I hate it when websites enforce this. Can we not instead enforce a lower limit on password length?

File: VotingApplication/VotingApplication.Web.Api/Views/Decision/Index.cshtml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                    <td class="negative">✘</td>
+                    <td class="affirmative">✔</td>
+                </tr>
+                <tr>
+                    <th colspan="3"></th>
+                </tr>
+                <tr>
+                    <td>Poll Details emailed to you</td>
+                    <td class="affirmative">✔</td>
+                    <td class="affirmative">✔</td>
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web.Api/Views/Decision/Index.cshtml]: Not entirely sure how excited a user would be about this 'feature'. Surely, this is a given?

File: VotingApplication/VotingApplication.Web.Api/Views/Decision/Index.cshtml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                    <td class="negative">✘</td>
+                    <td class="affirmative">✔</td>
+                </tr>
+                <tr>
+                    <td>Invitation-only Polls</td>
+                    <td class="negative">✘</td>
+                    <td class="affirmative">✔</td>
+                </tr>
+                <tr>
+                    <td>User can add Options</td>
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web.Api/Views/Decision/Index.cshtml]: Maybe rephrase this as 'Voters can add options'

File: VotingApplication/VotingApplication.Web.Api/Scripts/Create.js (hunk 1 above)
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web.Api/Scripts/Create.js]: This should be done on creation rather than submission to remove the surprise when its a 30 minute timelimit.

It looks like this is handled by the data-binding already though, so i'll remove it here

File: VotingApplication/VotingApplication.Web.Api/Views/Decision/Index.cshtml (hunk 4 above)
Comment (tpkelly) [inline VotingApplication/VotingApplication.Web.Api/Views/Decision/Index.cshtml]: Very rarely for polling websites, but also a feature I use a lot.

Plus I needed some more ticks for the Basic Poll column!
